src/platform/monitoring/DowntimeNotification/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+
+import { getScheduledDowntime } from '../actions';
+import { getSoonestDowntime } from '../util/helpers';
+import services from '../config/services';
+import serviceStatus from '../config/serviceStatus';
+import Down from '../components/Down';
+import DowntimeApproaching from '../components/DowntimeApproaching';
+
+/**
+ * React component used to conditionally render children components based on the status (down, down-approaching, or ok) of Vets.gov services.
+ * @property {string} [appTitle] - The name of the consuming application, which will be displayed in downtime messaging.
+ * @property {node} [children] - React components to be rendered based on downtime.
+ * @property {node} [content] - Alias for React.children.
+ * @property {Array<string>} dependencies - An array of services that the consuming application requires in order to operate.
+ * @property {function} getScheduledDowntime - [Provided by container] An action creator that retrieves the array of downtime from the API downtime endpoint.
+ * @property {boolean} isReady - [Provided by container] A flag for indicating whether the downtime array has been retrieved from the API and if the component can render.
+ * @property {Node} [loadingIndicator] - A React component that will be rendered while the request to the API for downtime information is pending.
+ * @property {function} [render] - A function that may be supplied for custom rendering, useful for customizing how downtime/downtime approaching is handled. Receives the derived status, downtimeWindow, downtimeMap, children as arguments.
+ * @module platorm/monitoring/DowntimeNotification
+ */
+class DowntimeNotification extends React.Component {
+
+  static propTypes = {
+    appTitle: PropTypes.string,
+    children: PropTypes.node,
+    content: PropTypes.node,
+    dependencies: PropTypes.arrayOf(PropTypes.oneOf(Object.values(services))).isRequired,
+    getScheduledDowntime: PropTypes.func.isRequired,
+    isReady: PropTypes.bool,
+    loadingIndicator: PropTypes.node,
+    render: PropTypes.func
+  };
+
+  componentDidMount() {
+    if (this.props.shouldSendRequest) this.props.getScheduledDowntime();
+  }
+
+  render() {
+    if (!this.props.isReady) {
+      return this.props.loadingIndicator || <LoadingIndicator message={`Checking the ${this.props.appTitle} status...`}/>;
+    }
+
+    const downtime = getSoonestDowntime(this.props.serviceMap, this.props.dependencies);
+    const children = this.props.children || this.props.content;
+
+    let content = children;
+
+    if (downtime) {
+      if (this.props.render) {
+        content = this.props.render(downtime, children);
+      } else if (downtime.status === serviceStatus.down) {
```

Comment:
Would be good to push these conditionals into `mapStateToProps` and simplify the component's props with explicit view behaviors. 

Could destructure `appTitle`, `startTime`, `endTime`, `showChildren` and `downtimeComponentType` from `this.props` and `downtimeComponentType ` could be an enum of something like `CURRENT`, `FUTURE`, and `PROVIDED`. 

There's no reason to implicitly control the showing of children with the downtime type- just make it an explicit prop and pass it from `mapStateToProps`. Then it becomes much clearer exactly what this component is doing just from the props. 

Would be good to flatten the arguments passed to `render` prop- i.e. instead of passing `downtime` which isn't well defined, pass `startTime`and `endTime` explicitly. If an object is preferable, then pass the arguments explicitly e.g. `render({ startTime, endTime }, children)`.

Reply:
Thanks for the suggestions. I'll see what I can do!